content/blog/2025-12-14-lel-final-project.md

```diff
@@ -0,0 +1,167 @@
++++
+title = "LaTeX Equation Language (LEL): LaTeX equation to LLVM"
+
+[extra]
+latex = false
+bio = """
+Ruolin Ye is a student in Advanced Compilers.
+"""
+
+[[extra.authors]]
+name = "Ruolin Ye"
++++
+
+# Intro
+LaTeX is a common language for writing professional documents like academic papers. In the past, I have worked on many research projects that involve implementing mathematical equations. It’s natural to think of a tool that converts the LaTeX formatted equations into usable code directly and allows the generated code to connect with other developing technologies.
+
+# Problem Statement
+This project asks: *can we automatically turn real LaTeX equations into executable code via a small compiler?*  
+I build a pipeline that:
+- normalizes LaTeX math from real sources (using MiTeX),
+- parses a LaTex Equation Language (LEL) supporting arithmetic, conditionals, and function calls, and
+- compiles each accepted equation to LLVM IR and checks it against a reference evaluator.
+
+By design, LEL only targets **scalar** expressions with simple function headers (`name(args) = expr`), and explicitly does *not* attempt to handle the full complexity of LaTeX: integrals, matrices, sums/products, asymptotic notation, or rich macro systems are all out of scope for this first prototype.
+
+# Implementation
+I will detail the implementation in this section. I will first introduce the datasets I use, explain the components and compare what I proposed vs. what's actually implemented, and then detail how I implemented each component.
+
+My code is open sourced at [link](https://github.com/YoruCathy/cs6120_final_project).
+## Dataset
```

Comment:
Maybe add a sentence here with an overview of the three datasets, and describing what you needed them for? (Namely, to validate your compiler's coverage of common LaTeX math constructs.)